core/mapserver/mapserver_test.go

```diff
@@ -131,9 +133,61 @@ func TestSetGet(t *testing.T) {
 			continue
 		}
 		for k, l := range resp2.MapLeafInclusion {
+			if got, want := l.Leaf.Index, tc.leaves[k].Index; !bytes.Equal(got, want) {
+				t.Errorf("GetLeaves(%v).Index[%v]: %s, want %s", i, k, got, want)
+			}
 			if got, want := l.Leaf.LeafValue, tc.leaves[k].LeafValue; !bytes.Equal(got, want) {
-				t.Errorf("GetLeaves(%v).Leaf[%v]: %s, want %s", i, k, got, want)
+				t.Errorf("GetLeaves(%v).LeafValue[%v]: %s, want %s", i, k, got, want)
 			}
 		}
 	}
 }
+
+func TestGetSignedMapRoot(t *testing.T) {
+	env, err := newEnv()
+	if err != nil {
+		t.Fatalf("Error creating env: %v", err)
+	}
+	defer env.db.Close()
+
+	for i, tc := range []struct {
+		epoch  int64
+		leaves []*trillian.MapLeaf
+	}{
+		{
+			epoch: 0,
+			leaves: []*trillian.MapLeaf{
+				&trillian.MapLeaf{Index: index(0), LeafValue: []byte("foo")},
+				&trillian.MapLeaf{Index: index(1), LeafValue: []byte("bar")},
+			}},
+		{
+			epoch: 1,
+			leaves: []*trillian.MapLeaf{
+				&trillian.MapLeaf{Index: index(0), LeafValue: []byte("foo1")},
+				&trillian.MapLeaf{Index: index(1), LeafValue: []byte("bar1")},
+			}},
+	} {
+		resp, err := env.m.SetLeaves(context.Background(), &trillian.SetMapLeavesRequest{
```

Comment:
nit: since we use this context twice, it can be defined at the beginning of the test function and then passed to `SetLeaves` and `GetSignedMapRoot`.